EventListener/TokenSubscriber.php

```diff
@@ -323,4 +333,138 @@ private function getCustomFieldValues(CustomObject $customObject, Token $token,
 
         return $fieldValues;
     }
+
+    public function onTokenReplacement(TokenReplacementEvent $event): void
+    {
+        $clickthrough = $event->getClickthrough();
+
+        if (!array_key_exists('dynamicContent', $clickthrough)) {
+            return;
+        }
+
+        $lead      = $event->getLead();
+        $tokenData = $clickthrough['dynamicContent'];
+
+        foreach ($tokenData as $data) {
+            $filterContent = $data['content'];
+
+            $isCustomObject = false;
+            foreach ($data['filters'] as $filter) {
+                $customFieldValues = $this->getCustomFieldDataForLead($filter['filters'], (int) $lead['id']);
+
+                if (!empty($customFieldValues)) {
+                    $isCustomObject = true;
+                    $lead           = array_merge($lead, $customFieldValues);
+                }
+
+                if ($isCustomObject && $this->matchFilterForLead($filter['filters'], $lead)) {
+                    $filterContent = $filter['content'];
+                    break;
+                }
+            }
+
+            if ($isCustomObject) {
+                $event->addToken('{dynamiccontent="'.$data['tokenName'].'"}', $filterContent);
+            }
+        }
+    }
+
+    /**
+     * @param array<mixed> $filters
+     *
+     * @return array<mixed>
+     */
+    private function getCustomFieldDataForLead(array $filters, int $leadId): array
```

Comment:
As the ID is BIGINT UNSIGNED which is higher than PHP's INT max value, we should work with it as string.
```suggestion
    private function getCustomFieldDataForLead(array $filters, string $leadId): array
```
I know it's not considered in all places, but we shouldn't add more that will have to be refactored later on.